### PR TITLE
feat(build.bat): customize PLATFORM_TOOLSET settings

### PR DESCRIPTION
**Pull request by lotem**
Similar to https://github.com/rime/librime/pull/178

Changes:

  - introduces `PLATFORM_TOOLSET` variable (defaults to `v140_xp`)
  - renames `BOOST_HOME` to `BOOST_ROOT` (to be consistent with `librime/build.bat`)

### CODE REVIEW

**Comment by lotem** [comment]
The check failed because I did not set the environment variable `PLATFORM_TOOLSET` in `appveyor.yml`. When I did to my local build, it worked.

The same environment variable also worked for several projects in `librime/thirdparty/src/` with this change https://github.com/rime/librime/pull/178 . No `*.props` file was needed to introduce the variable to project files.

However, I did not expect user to set the environment variable, but to define the macro in `weasel.props`.
Somehow the default value set in `weasel[-appveyor].props` did not take effect.

**Comment by nameoverflow** [comment]
Why not create a .props file in appveyor.install.bat ?

**Comment by Prcuvu** [comment]
It turns out that `weasel.props` must be imported before using any of its macros.

File: build.bat [hunk 1]
```diff
@@ -78,9 +78,22 @@ if errorlevel 1 goto error
 goto end
 
 :build_boost
-set boost_build_flags=toolset=msvc-14.0 variant=release link=static threading=multi runtime-link=static
-set boost_libs=--with-date_time --with-filesystem --with-locale --with-regex --with-signals --with-system --with-thread
-cd %BOOST_ROOT%
+
+set boost_build_flags=toolset=msvc-%VisualStudioVersion%^
```
Comment (lotem) [inline build.bat]: Should not use %VisualStudioVersion% here.
The number is platform toolset, not Visual Studio version.
For example msvc-14.1 for Visual Studio 15 2017.

https://stackoverflow.com/questions/42730478/version-numbers-for-visual-studio-2017-boost-and-cmake/42738533#42738533

**Comment by lotem** [comment]
Well, it's more complicated than what I thought.
It's great the properties file works as expected now with @Prcuvu's fix.
We might also need to set environment variables since
`build.bat boost` requires `BOOST_ROOT` and `BJAM_TOOLSET` set in CMD.
User have to set the environment variables manually at first run. Maybe it's a good idea to generate `weasel.props` to avoid typing those values twice.
It'll be nice not having to set environment variables afterwards, working in Visual Studio without command line.

**Comment by Prcuvu** [comment]
Typing toolset version in both build script and property sheet is inevitable since we do not build boost in Visual Studio IDE. It is a one-time job after all.

**Comment by lotem** [comment]
Wait... I have more code to commit.

**Comment by Prcuvu** [comment]
@lotem I would like to leave the job to users. If you come up with some better approach, force update is welcome.

**Comment by lotem** [comment]
Can we sync submodule librime to https://github.com/rime/librime/tree/rime-1.2.10 ?
With latest librime, environment settings in `weasel\env.bat` works for both `weasel\build.bat` and `librime\build.bat`.

**Comment by Prcuvu** [comment]
Currently AppVeyor CI downloads librime binaries of its latest commit, which is ugly and causes problems.

I'm planning to build librime in weasel's CI install stage and cache built binaries.

**Comment by lotem** [comment]
@Prcuvu 
> is ugly and problematic
Why do you think it's ugly? Can you share more details on encountered problems with this approach?

I'm thinking about splitting `librime` into smaller projects to save CI build time by reusing artifacts from dependency projects.

**Comment by Prcuvu** [comment]
* I say it's ugly because I have to modify `appveyor.install.bat` and add new artifact download URL every time I upgrade submodule version. Modification of `appveyor.install.bat` also triggers boost rebuild.
* If we always download artifact from latest commit, it will break weasel if librime of that commit happens to have a bug.

I would like to add a librime build cache, whose dependency tracks submodule change (if possible), into `appveyor.yml`.

**Comment by Prcuvu** [comment]
Well, reusing artifacts is a more economic approach. I will just specify artifact URL every time librime changes.

Feel free to merge when you think it is ready.

**Comment by lotem** [comment]
There is one more thing:
`librime/build/include` is missing in local build, and causes `weasel/build.bat` to fail.
Though the appveyor script handles this, it's still broken for user.
For that I'll update submodule `librime` to include this change
https://github.com/rime/librime/commit/06c9e8653f89407d6c4ee0d662f2da6117e98780
(Testing...)

**Pull request closed without merging**